pkg/apis/serving/v1beta1/route_validation.go

```diff
@@ -127,13 +127,33 @@ func (tt *TrafficTarget) Validate(ctx context.Context) *apis.FieldError {
 		errs = errs.Also(apis.ErrMissingOneOf(
 			"revisionName", "configurationName"))
 	}
+	return errs
+}
 
+func (tt *TrafficTarget) validateTrafficPercentage(errs *apis.FieldError) *apis.FieldError {
 	// Check that the traffic Percentage is within bounds.
 	if tt.Percent < 0 || tt.Percent > 100 {
 		errs = errs.Also(apis.ErrOutOfBoundsValue(
 			tt.Percent, 0, 100, "percent"))
 	}
+	return errs
+}
 
+func (tt *TrafficTarget) validateLatestRevision(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+	if apis.IsInSpec(ctx) && tt.LatestRevision != nil {
+		lr := *tt.LatestRevision
+		pinned := tt.RevisionName != ""
+		if pinned == lr {
+			// The senses for whether to pin to a particular revision or
```

Comment:
I am not sure I can understand the comment here.

Reply:
Happy to change the comment, but this is coming from the existing source: https://github.com/knative/serving/pull/3540/files/028796d15f5c41d67c225a8d3117cfc933386e57#diff-6af5a4f31c34c23000270ca77ea99320L86